pkg/query-service/app/logs/parser.go

```diff
@@ -247,13 +249,33 @@ func replaceInterestingFields(allFields *model.GetFieldsResponse, queryTokens []
 }
 
 func replaceFieldInToken(queryToken string, selectedFieldsLookup map[string]model.LogField, interestingFieldLookup map[string]model.LogField) (string, error) {
+	op := strings.TrimSpace(operatorRegex.FindString(queryToken))
+	opLower := strings.ToLower(op)
+
 	col, err := parseColumn(queryToken)
 	if err != nil {
 		return "", err
 	}
 
 	sqlColName := *col
 	lowerColName := strings.ToLower(*col)
+
+	if opLower == "exists" || opLower == "nexists" {
+		// creating the query token here as we have the metadata
+		field := model.LogField{}
+		var result string
+		if sfield, ok := selectedFieldsLookup[sqlColName]; ok {
+			field = sfield
+		} else if ifield, ok := interestingFieldLookup[sqlColName]; ok {
+			field = ifield
+		}
+		result = fmt.Sprintf("has(%s_%s_key, '%s')", field.Type, strings.ToLower(field.DataType), field.Name)
```

Comment:
Yeah will have to handle the top level keys

Reply:
@srikanthccv , this has the same problem that we have discussed here https://github.com/SigNoz/signoz/issues/2592.

For this I was thinking 

```
// handle static fields, timestamp and id is not required but added them regardless
defValue := ""
if lowerColName == "trace_id" || lowerColName == "span_id" || lowerColName == "severity_text" || lowerColName == "id" {
	defValue = "''"
}
if lowerColName == "trace_flags" || lowerColName == "severity_number" || lowerColName == "timestamp" {
	defValue = "0"
}
result := ""
if opLower == "exists" {
	result = fmt.Sprintf("%s!=%s", sqlColName, defValue)
} else {
	result = fmt.Sprintf("%s=%s", sqlColName, defValue)
}
```

So for all the string types I am checking empty string and all the number types I can check !=0 since both trace_flags and severity_number values are > 0 cc @ankitnayan

Reply:
This seems reasonable to me.

Reply:
I have added the changes please have a look once